dll/win32/ws2help/ws2help.spec

```
@@ -1,23 +1,47 @@
+@ stdcall -version=0x400-0x503 WahDestroyHandleContextTable(ptr)
+@ stdcall -version=0x400-0x503 WahDisableNonIFSHandleSupport()
+@ stdcall -version=0x400-0x503 WahEnableNonIFSHandleSupport()
+@ stdcall -version=0x400-0x503 WahEnumerateHandleContexts(ptr ptr ptr)
+@ stdcall -version=0x400-0x503 WahInsertHandleContext(ptr ptr)
+@ stdcall -version=0x400-0x503 WahNotifyAllProcesses(long)
+@ stdcall -version=0x400-0x503 WahOpenApcHelper(ptr)
+@ stdcall -version=0x400-0x503 WahOpenCurrentThread(long ptr)
+@ stdcall -version=0x400-0x503 WahOpenHandleHelper(ptr)
+@ stdcall -version=0x400-0x503 WahOpenNotificationHandleHelper(ptr)
+@ stdcall -version=0x400-0x503 WahQueueUserApc(long ptr ptr ptr)
+@ stdcall -version=0x400-0x503 WahReferenceContextByHandle(ptr long)
+@ stdcall -version=0x400-0x503 WahRemoveHandleContext(ptr ptr)
+@ stdcall -version=0x400-0x503 WahWaitForNotification(long long ptr ptr)
+
+@ stdcall -version=0x600+ WahCloseApcHelper(long) ws2_32.WahCloseApcHelper
+@ stdcall -version=0x600+ WahCloseHandleHelper(long) ws2_32.WahCloseHandleHelper
+@ stdcall -version=0x600+ WahCloseNotificationHandleHelper(long) ws2_32.WahCloseNotificationHandleHelper
+@ stdcall -version=0x600+ WahCloseSocketHandle(long ptr) ws2_32.WahCloseSocketHandle
+@ stdcall -version=0x600+ WahCloseThread(long ptr) ws2_32.WahCloseThread
+@ stdcall -version=0x600+ WahCompleteRequest(long ptr ptr long long) ws2_32.WahCompleteRequest
+@ stdcall -version=0x600+ WahCreateHandleContextTable(ptr) ws2_32.WahCreateHandleContextTable
+@ stdcall -version=0x600+ WahCreateNotificationHandle(long ptr) ws2_32.WahCreateNotificationHandle
+@ stdcall -version=0x600+ WahCreateSocketHandle(long ptr) ws2_32.WahCreateSocketHandle
+@ stdcall -version=0x600+ WahDestroyHandleContextTable(ptr) ws2_32.WahDestroyHandleContextTable
+@ stdcall -version=0x600+ WahDisableNonIFSHandleSupport() ws2_32.WahDisableNonIFSHandleSupport
+@ stdcall -version=0x600+ WahEnableNonIFSHandleSupport() ws2_32.WahEnableNonIFSHandleSupport
+@ stdcall -version=0x600+ WahEnumerateHandleContexts(ptr ptr ptr) ws2_32.WahEnumerateHandleContexts
+@ stdcall -version=0x600+ WahInsertHandleContext(ptr ptr) ws2_32.WahInsertHandleContext
+@ stdcall -version=0x600+ WahNotifyAllProcesses(long) ws2_32.WahNotifyAllProcesses
+@ stdcall -version=0x600+ WahOpenApcHelper(ptr) ws2_32.WahOpenApcHelper
+@ stdcall -version=0x600+ WahOpenCurrentThread(long ptr) ws2_32.WahOpenCurrentThread
+@ stdcall -version=0x600+ WahOpenHandleHelper(ptr) ws2_32.WahOpenHandleHelper
+@ stdcall -version=0x600+ WahOpenNotificationHandleHelper(ptr) ws2_32.WahOpenNotificationHandleHelper
+@ stdcall -version=0x600+ WahQueueUserApc(long ptr ptr ptr) ws2_32.WahQueueUserApc
+@ stdcall -version=0x600+ WahReferenceContextByHandle(ptr long) ws2_32.WahReferenceContextByHandle
+@ stdcall -version=0x600+ WahRemoveHandleContext(ptr ptr) ws2_32.WahRemoveHandleContext
+@ stdcall -version=0x600+ WahWaitForNotification(long long ptr ptr) ws2_32.WahWaitForNotification
```

```suggestion
@ stdcall -version=0x600+ WahWaitForNotification(long long ptr ptr) ws2_32.WahWaitForNotification
@ stdcall -version=0x600+ WahWriteLSPEvent() ws2_32.WahWriteLSPEvent
```
In principle, WahWriteLSPEvent is there and also redirected.
Other question: Because these 0x600+ are redirections, is it necessary to copy their list of parameters as well?

---

Yes, parameters are needed to generate the correct symbols in the import lib.